README-DEV.md

```diff
@@ -126,3 +126,32 @@ This workflow is used to check that modifications to `*.json` files are in line
 #### Initial Checksum Creation: `generate-initial-checksums.yml`
 
 This `workflow_dispatch`-triggered workflow generates checksums of a given config branch, and optionally commits them. This is useful for generating checksums for an entirely new config branch, so the workflows above have something to compare against.
+
+### CI Configuration File
+
+This is the `config/ci.json` configuration file for specifying different test markers, or test versions based on type of the test to run, and the name of the git branch or tag. The different types of test are defined as:
+- `scheduled`: These are scheduled monthly reproducibility tests that are run as part of [`schedule-2-start.yml`](#repro-ci-1). The keys under these tests represent released config tags to run scheduled checks on.
+- `reproducibility`: These are reproducibility tests are run as part of [`pr-1-ci.yml`](#repro-ci). The keys under these tests represent the target branches into which pull requests are being merged.
+- `qa` - These are quick QA tests are run as part of [`pr-1-ci.yml`](#qa-ci). The keys under these tests represent the target branches into which pull requests are being merged.
+
+As most of the tests use the same test and python versions, and similar markers, there are two levels of defaults. There's a default at test type level which is useful for defining test markers - this selects certain pytests to run in `model-config-tests`. There is an outer global default, which is used if a property is not defined for a given branch/tag and it is not defined for the test default.
+
+#### parse-ci-config
+
+This action parses the CI testing configuration file. The caller of the action needs to checkout the branch where the config file is defined before running this action.
+
+#### Inputs
+
+| Name | Type | Description | Required | Example |
+| ---- | ---- | ----------- | -------- | ------- |
+| check | `string` | The type of check/test to run | true | `scheduled` |
+| branch-or-tag | `string` | The name of git branch or tag | true | 'release-1deg_jra55_ryf-2.0' |
+| config-filepath | `string` | Path to configuration file | true | `config/ci.json` |
+
+#### Outputs
+
+| Name | Type | Description |  Example |
+| ---- | ---- | ----------- | -------- |
+| markers | `string` | Markers used for the pytest checks, in the python format | `checksum` |
+| model-config-tests-version | `string` | The version of the model-config-tests | `0.0.1` |
+| python-version | `string` | The python version used to create test virtual environment | `3.11.0` |
```

Comment:
I reckon it would be a good idea to add this information to a `.github/actions/parse-ci-config/README.md`, much like the actions in https://github.com/ACCESS-NRI/actions/. For example, https://github.com/ACCESS-NRI/actions/tree/main/.github/actions/pr-comment has an associated README.md with the action.

Reply:
Good point, I've now moved it to it's own readme